src/plugins/validate-semantic/validators/oas3/refs.js

```diff
@@ -6,26 +6,58 @@ export const validateOAS3RefsForRequestBodiesReferenceRequestBodyPositions = ()
         const value = node.node
         const ref = value.$ref
 
-        if(!ref) {
+        if (!ref) {
           return acc
         }
 
         const [, refPath = ""] = ref.split("#")
         const pathArr = refPath.split("/") || []
         const parentRefKey = pathArr.slice(-2)[0]
         const targetRefKey = pathArr.slice(-1)[0]
-        if(
-          targetRefKey !== "requestBody"
-          && parentRefKey !== "requestBody"
-          && parentRefKey !== "requestBodies"
-        ) {
+
+        // External references starting with http:// or https://
+        if (ref.startsWith("https://") || ref.startsWith("http://")){
+          return acc  
+        }
+
+        // External references starting with ./
+        if (ref.startsWith("./")){
+          return acc  
+        }
+
+        // Extensions are valid
+        if (ref.startsWith("#/") && (parentRefKey.startsWith("/x-") || targetRefKey.startsWith("/x-"))){
+          return acc  
+        }
+
+        // Edge cases
+        if (ref.endsWith("requestBody") && (ref.startsWith("#/paths") || ref.startsWith("#/components"))){
+          return acc  
+        }
+
+        // External references ending with yaml
+        if (ref.endsWith(".yaml")) {
+          return acc
+        }
```

Comment:
@ralphdoe all checks for external $refs (`http(s)://`, `./`, `.yaml`) can be replaced with
```js
// Ignore external references
if (!ref.startsWith("#/") {
  return acc
}
```
or
```js
if (ref.startsWith("#/")) {
  // validations for local $refs go here...
  if (...) {...}
}
```
This logic will automatically exclude all non-local $refs such as
```
http(s)://...
./path/to/...
../path/to/...
file.json
subfolder/file.yaml
etc.
```

Reply:
Done.